src/main/scala/edu/berkeley/cs/rise/opaque/execution/operators.scala

```diff
@@ -197,9 +200,10 @@ case class EncryptedProjectExec(projectList: Seq[NamedExpression], child: SparkP
   override def executeBlocked(): RDD[Block] = {
     val projectListSer = Utils.serializeProjectList(projectList, child.output)
     timeOperator(child.asInstanceOf[OpaqueOperatorExec].executeBlocked(), "EncryptedProjectExec") {
-      childRDD => childRDD.map { block =>
+      childRDD => 
+        childRDD.map { block =>
         val (enclave, eid) = Utils.initEnclave()
-        Block(enclave.Project(eid, projectListSer, block.bytes))
+        Block(enclave.Project(eid, projectListSer, block.bytes, TaskContext.getPartitionId))
```

Comment:
Why are you using `TaskContext.getPartitionId` here instead of `mapPartitionsWithIndex`?

Also, with either of those two choices, how does the worker enclave get the partition ID? Unless it's encrypted, I think the partition ID could also be changed by the server side's untrusted code, so we'd be relying on more than just the trusted driver.